tests/json_cart_parser_file_test.py

```diff
@@ -0,0 +1,210 @@
+    expected_content = json.dumps(valid_json_data_2real_2virtual_items, cls=CartEncoder)
+    test_file.close()
+    # to remove created file with data
+    # os.remove(test_file_path)
+    assert actual_content == expected_content, f"Cart data writen in created file doesn't mach expected data, Actual data: {actual_content}, Expected data: {expected_content}"
+
+
+@pytest.mark.parametrize("invalid_cart_data", [
+        '{"invalid_cart_data"}',
+        '{"invalid_key": "value"}',
+        '{"cart_name": "TestCart",  "real_items": [] }',
+        '{"cart_name": "TestCart",  "virtual_items": [] }',
+        '{"cart_name": "TestCart", "real_items": [{"weight": 2.0, "name": "Toy"}], "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [{"weight": 2.0, "price": 100.3}], "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [ {"name": "Toy", "price": 100.3}],  "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [{"disk_size": 8500.0, "name": "Microsoft office"}], "total": 1200.1]}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [{"disk_size": 8500.0, "price": 30.0}], "total": 1200.1]}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [{"name": "Microsoft office", "price": 30.0}], "total": 1200.1]}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [{"disk_size": 8500.0, "name": "Microsoft office", "price": 30.0}]}',
+        '{"cart_name": "TestCart", "real_items": [{"weight": 2.0, "name": "Toy", "price": 100.3}, {"weight": 1.1, "name": "Laptop"}], "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [{"disk_size": 8500.0, "name": "Microsoft office", "price": 30.0}, {"disk_size": 9500.0, "name": "Microsoft office"}], "total": 1200.1}',
+        '{"cart_name": Test, "real_items": [], "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": {"weight": "2.0", "name": "Toy", "price": 100.3}, "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": {"weight": 2.0, "name": Toy, "price": 100.3}, "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": {"weight": 2.0, "name": Toy, "price": "100.3"}, "virtual_items": [], "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": {"disk_size": "8500.0", "name": "Microsoft office", "price": 30.0}, "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": {"disk_size": 8500.0, "name": Microsoft office, "price": 30.0}, "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": {"disk_size": 8500.0, "name": "Microsoft office", "price": "30.0"}, "total": 1200.1}',
+        '{"cart_name": "TestCart", "real_items": [], "virtual_items": [], "total": test}',
+])
+
```

Comment:
don't need to put 2 lines between test and parametrize decorator

Reply:
ok, it is done